app/controllers/maestros_controller.rb

```diff
@@ -0,0 +1,53 @@
+class MaestrosController < ApplicationController
+
+  before_action :find_maestro, only: [:show, :edit, :update, :destroy]
```

Comment:
`before_action`s, while being useful and reducing code, add a layer of magic in Rails that I tend to stay away from. Instead of magically having an instance variable defined on, for example, the `show` action, I will explicitly set the instance variable in it, but abstract away the logic. For example:
```ruby
def show
  @maestro = find_maestro
end

# ...

private

def find_maestro
  Maestro.find(params[:id])
end
```